scrapers/distgit.py

```diff
@@ -42,16 +44,50 @@ def run(self, since=None, until=None):
             end_date = timestamp_to_date(until)
         results = self.get_distgit_data(start_date, end_date)
         log.info('Successfully fetched {0} results from Teiid'.format(len(results)))
-        self.update_neo4j(results)
+        # Upload the results to Neo4j
+        _update_neo4j_partial = partial(
+            self._update_neo4j, neomodel_config.DATABASE_URL, len(results))
+        pool = Pool(2)
+        pool.map(_update_neo4j_partial, self._get_result_chunks(results))
         log.info('Initial load of dist-git commits and pushes complete!')
 
-    def update_neo4j(self, results):
+    @staticmethod
+    def _get_result_chunks(results):
+        """
+        Yield a tuple with a counter and chunk of results.
+
+        :param list results: a list of dictionaries representing results from Teiid to chunk up
+        :return: generator of a tuple with a counter and chunk of results
+        :rtype: generator
+        """
+        list_iterator = iter(results)
+        chunk_size = 10000
+        count = 0
+        while True:
+            chunk = list(islice(list_iterator, chunk_size))
+            if chunk:
+                yield (count, chunk)
+            else:
+                break
+            count += chunk_size
+
+    @staticmethod
+    def _update_neo4j(neo4j_url, total_results, counter_and_results):
         """
-        Update Neo4j with the dist-git commit and push information from Teiid.
+        Update Neo4j results via mapping with multiprocessing.
 
-        :param list results: a list of dictionaries
+        :param str neo4j_url: database url for Neo4j
+        :param int total_results: the total number of results that will be processed. This is used
+        for a logging statement about progress.
+        :param tuple counter_and_results: a tuple where the first index is the current counter and
+        the second index is a list of dictionaries representing results from Teiid
         """
-        pool = Pool(processes=8)
+        previous_total = counter_and_results[0]
+        results = counter_and_results[1]
+        # Since _update_neo4j will be run in a separate process, we must configure the database
+        # URL every time the method is run.
+        neomodel_config.DATABASE_URL = neo4j_url
+        pool = ThreadPool(8)
```

Comment:
Since we are using multi-processing above, this pool should be reduced to 4. Having 16 threads banging against cgit may be too much.